sys/kern/subr_prf.c

```diff
@@ -443,7 +444,7 @@ prf_putbuf(char *bufr, int flags, int pri)
 		msgbuftrigger = 1;
 	}
 
-	if (flags & TOCONS) {
+	if ((flags & TOCONS) && (!cn_mute)) {
 		if ((!KERNEL_PANICKED()) && (constty != NULL))
 			msgbuf_addstr(&consmsgbuf, -1,
 			    bufr, /*filter_cr*/ 0);
```

Comment:
So why here, and before, do you not add it do msgbuf?

Reply:
It already exists there, but some things bypass calls, and call this directly.  (Specifically other kernel modules, including the ELF loader and a few other places.)  Placing the check here, at the lowest level, guarantees that anything calling for console output will be intercepted.